### DISCUSSION TITLE
calculateDiffMeth timing out on Mox - do I need to run R Studio server?

### DISCUSSION THREAD

**Discussion by yaaminiv** [RobertsLab/resources#1188] [Q&A] [answered]
I can finally run R scripts on mox!...but the `calculateDiffMeth` command within `methylKit` is timing out. I first called [this R script](https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/code/06-methylKit.R) from [this SLURM script](https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/code/06-methylKit-SLURM.sh). This is the `calculateDiffMeth` code:

```
differentialMethylationStatsTreatmentFem <- methylKit::calculateDiffMeth(methylationInformationFilteredCov5Fem, covariates = covariateMetadataFem, overdispersion = "MN", test = "Chisq")
```

My script ran, but it timed out in ~5 hours with the following error when it hit the `calculateDiffMeth` code:

![Screen Shot 2021-04-27 at 9 33 51 AM](https://user-images.githubusercontent.com/22335838/116278774-ae0e1680-a73b-11eb-915d-6a16f7173730.png)

I couldn't see the warnings since I wasn't running the script interactively, so I thought I should try an interactive build node. I then opened a build node for 4 hours, loaded the R data I saved, and ran the the same `calculateDiffMeth` code. The job was force terminated and I didn't see any warnings.

![Screen Shot 2021-04-27 at 9 33 10 AM](https://user-images.githubusercontent.com/22335838/116278685-96cf2900-a73b-11eb-8495-961555c4c245.png)

@kubu4 @sr320 Do you think it's worth trying this command on the R Studio server, at least to investigate the warnings? 

*edited to include links*

**Comment by kubu4** [comment]
>  Do you think it's worth trying this command on the R Studio server,

Sure!

**Comment by yaaminiv** [comment, marked as answer]
> The job was force terminated and I didn't see any warnings.

The job was likely force terminated since I only requested a build node for 4 hours

> My script ran, but it timed out in ~5 hours with the following error when it hit the `calculateDiffMeth` code

...turns out it was a typo in my code the entire time, and not a function of running R SLURM script, since I got the [same error with R Studio server](https://github.com/RobertsLab/resources/discussions/1190#discussioncomment-671472) :grimacing: :expressionless: :sweat_smile:

At least I can fix the code as I go with R Studio Server!

*edited to properly quote previous error message descriptions*
